ExampleMod/Content/Items/Weapons/ExampleDualUseWeapon

```diff
@@ -0,0 +1,103 @@
+using ExampleMod.Tiles;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Items.Weapons
+{
+	  public class ExampleDualUseWeapon : ModItem
+	  {
+		  public override void SetStaticDefaults() {
+			Tooltip.SetDefault("This weapon does something special with <right>.");
+		}
+		public override void SetDefaults() {
+      item.damage = 50;
+			item.melee = true;
+			item.width = 40;
+			item.height = 40;
+			item.useTime = 20;
+			item.useAnimation = 20;
+			item.useStyle = ItemUseStyleID.SwingThrow;
+			item.knockBack = 6;
+			item.value = 10000;
+			item.rare = ItemRarityID.Green;
+			item.UseSound = SoundID.Item1;
+			item.autoReuse = true;
+			item.shoot = ProjectileID.Bee;
+			item.shootSpeed = 5f;
+		}
+
+		public override void AddRecipes() {
+      CreateRecipe()
+				.AddIngredient<ExampleItem>()
+				.AddTile<ExampleWorkbench>()
+				.Register();
+		}
+
+		public override bool AltFunctionUse(Player player) {
+      // Returned true so you can use the AltFunctionUse
+			return true;
+		}
+
+		public override bool CanUseItem(Player player) {
+    // Check for when the player is using the AltFunctionUse
+			if (player.altFunctionUse == 2) {
+				item.useStyle = ItemUseStyleID.Stabbing;
+				item.useTime = 20;
+				item.useAnimation = 20;
+				item.damage = 50;
+				item.shoot = ProjectileID.Bee;
+			}
+			else {
+      // If not using the AltFunctionUse, use your weapon's default stats, this else block is important because the changes wouldn't revert otherwise
+				item.useStyle = ItemUseStyleID.SwingThrow;
+				item.useTime = 40;
+				item.useAnimation = 40;
+				item.damage = 100;
+				item.shoot = ProjectileID.None;
+			}
+			return base.CanUseItem(player);
+		}
+    // Called upon hitting any NPC
+		public override void OnHitNPC(Player player, NPC target, int damage, float knockBack, bool crit) {
+			if (player.altFunctionUse == 2) {
+				target.AddBuff(BuffID.Ichor, 60);
+			}
+			else {
+				target.AddBuff(BuffID.OnFire, 60);
+			}
+		}
+
+		public override void MeleeEffects(Player player, Rectangle hitbox) {
+    // All of this is called during the duration of the weapon's useAnimation
+			if (Main.rand.NextBool(3)) {
+				if (player.altFunctionUse == 2) {
+					int dust = Dust.NewDust(new Vector2(hitbox.X, hitbox.Y), hitbox.Width, hitbox.Height, 169, 0f, 0f, 100, default(Color), 2f);
+					Main.dust[dust].noGravity = true;
+					Main.dust[dust].velocity.X += player.direction * 2f;
+					Main.dust[dust].velocity.Y += 0.2f;
+				}
+				else {
+					int dust = Dust.NewDust(new Vector2(hitbox.X, hitbox.Y), hitbox.Width, hitbox.Height, DustID.Fire, player.velocity.X * 0.2f + (float)(player.direction * 3), player.velocity.Y * 0.2f, 100, default(Color), 2.5f);
+					Main.dust[dust].noGravity = true;
+				}
+			}
+    }
+    // Give some cool Shoot effects
```

Comment:
Describe what the math does please. "take the length of the vector (speedX, speedY) to get total travel speed, then put that speed on the X axis so it always shoots flat." Alternatively it might be easier to get the speed from a projectile property?